src/main.tsx

```diff
@@ -1,10 +1,13 @@
+import { LocalStorageProvider } from '@/contexts/LocalStorageContext';
 import React from 'react';
 import ReactDOM from 'react-dom/client';
 import App from './App.tsx';
 import './index.scss';
 
 ReactDOM.createRoot(document.getElementById('root') as HTMLElement).render(
   <React.StrictMode>
-    <App />
+    <LocalStorageProvider>
+      <App />
+    </LocalStorageProvider>
```

Comment:
Move this Context to App.tsx

```suggestion
    <App />
```